geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelper.java

```diff
@@ -0,0 +1,176 @@
+      throws InterruptedException {
+    final int numThreads = runnables.size();
+    final List<Throwable> exceptions = Collections.synchronizedList(new ArrayList<Throwable>());
+    final ExecutorService threadPool = Executors.newFixedThreadPool(numThreads);
+
+    List<Callable<Object>> callables = new ArrayList<>();
+    for (Runnable runnable : runnables) {
+      callables.add(toCallable(runnable));
+    }
+
+    List<Future<Object>> futures = new ArrayList<>();
+    try {
+      futures.addAll(threadPool.invokeAll(callables, timeoutSeconds, TimeUnit.SECONDS));
+    } catch (InterruptedException e) {
+      threadPool.shutdownNow();
+      exceptions.add(e);
+    }
+
+    for (Future<Object> future : futures) {
+      try {
+        future.get();
+        exceptions.add(null);
+      } catch (CancellationException e) {
+        exceptions.add(e);
+      } catch (InterruptedException e) {
+        throw e;
+      } catch (ExecutionException e) {
+        exceptions.add(e.getCause());
+      }
+    }
+
+    return exceptions;
+  }
+
+  /**
+   * Runs a List of Callables in a thread pool with a default timeout of 300 seconds. Returns a List
+   * of results after the completion of all threads, or once the timeout is reached, whichever
+   * happens first.
+   *
+   * @param callables of the same return type
+   * @param <T>
+   * @return List of results of the same type as the return statements of the Callables given as
+   *         input. If there are no exceptions thrown in the threads then the list of results will
+   *         be in the
+   *         order of the callables passed in. If there are exceptions, the stacktrace of the
+   *         exceptions
+   *         will be printed and a RuntimeException will be thrown.
+   * @throws RuntimeException if any of the threads throw ExecutionException or
+   *         CancellationException
+   */
+  public static <T> List<T> runCallables(List<Callable<T>> callables) throws InterruptedException {
+    return runCallables(callables, 30);
+  }
+
+  /**
+   * Runs a List of Callables in a thread pool with a timeout in Seconds. Returns a List
+   * of results after the completion of all threads, or once the timeout is reached, whichever
+   * happens first.
+   *
+   * @param callables of the same return type
+   * @param <T>
+   * @return List of results of the same type as the return statements of the Callables given as
+   *         input. If there are no exceptions thrown in the threads then the list of results will
+   *         be in the
+   *         order of the callables passed in. If there are exceptions, the stacktrace of the
+   *         exceptions
+   *         will be printed and a RuntimeException will be thrown.
+   * @throws RuntimeException if any of the threads throw ExecutionException or
+   *         CancellationException
+   */
+  public static <T> List<T> runCallables(List<Callable<T>> callables, Integer timeoutSeconds)
+      throws InterruptedException {
```

Comment:
Would it be useful to return the results of the Callables even if one or more of them threw an exception? Also, the list of exceptions should identify which callable the exception is associated with. In the IntegrationTest included with this PR, the tests that have more than one callable throwing an exeception don't check if the excpetions came from the expected callables.